app/src/verifier.h

```diff
@@ -362,17 +363,106 @@ namespace scitt::verifier
         }
         else if (phdr.issuer.has_value())
         {
-          // IETF SCITT claim
-          key = process_ietf_profile(
-            phdr, tx, current_time, resolution_cache_expiry, configuration);
-
-          try
+          if (phdr.issuer->starts_with("did:x509"))
           {
-            cose::verify(data, key);
+            // IETF SCITT did:x509 claim
+            if (phdr.x5chain.has_value())
+            {
+              // NB: In later revisions of SCITT, x5chain is unprotected, and
+              // only x5t is. This logic will need to authenticate x5chain[0]
+              // against x5t before it can proceed to verify the signature.
+              // Verify the signature as early as possible
+              OpenSSL::Unique_X509 leaf =
+                parse_certificate(phdr.x5chain.value()[0]);
+              key = PublicKey(leaf, std::nullopt);
+              try
+              {
+                cose::verify(data, key);
+              }
+              catch (const cose::COSESignatureValidationError& e)
+              {
+                throw VerificationError(e.what());
+              }
+              // Then authenticate the did:x509 claim against the x5chain
+              std::string pem_chain;
+              for (auto const& c : phdr.x5chain.value())
+              {
+                pem_chain += ccf::crypto::cert_der_to_pem(c).str();
+              }
+              auto did_document_str = didx509::resolve(
```

Comment:
Provided the chain will be verified earlier you could just use `create_did_document` instead. But it would be better to avoid that altogether to remove unnecessary json create/parse. This might otherwise cap the amount of transaction we will be able to process during the heavy load

Reply:
I think what we could do is have didx509::resolve() return a C++ object as well/instead of the JSON document, so we can skip the serialisation/parse. `create_did_document` would not help, because the things `resolve()` ((index, digest) and eku) needs to check are not present in the document and are only in the x5chain.

Reply:
Looked more closely, and I think we can probably simplify the resolve() to a resolve_jwk() that returns a single key, and skip parsing the outer document we don't need here. Let's discuss.

Reply:
As discussed, logged as an optimisation in the did:x509 repo: https://github.com/microsoft/didx509cpp/issues/16